src/server/transaction.cc

```diff
@@ -837,9 +837,9 @@ void Transaction::ExecuteAsync() {
   // IsArmedInShard in other threads.
   run_count_.store(unique_shard_cnt_, memory_order_release);
 
-  // Execute inline when we can. We can't use coordinator_index_ because we may offload this
-  // function to run in a different thread.
-  if (unique_shard_cnt_ == 1 && ServerState::tlocal()->thread_index() == unique_shard_id_) {
```

Comment:
This would actually have caused a bug for non atomic squashing

Reply:
Nice catch! Too bad this is already in the released version :(

Reply:
I think we disabled squashing anyway in this version anyway.

Reply:
It doesn't cause, it would have caused if the latest pr was merged